hack/e2e/run-rp-and-e2e.sh

```diff
@@ -124,6 +130,10 @@ clean_e2e() {
     rm -f $KUBECONFIG
 }
 
+set_az_cli_sub(){
+    az account set -s $AZURE_SUBSCRIPTION_ID
```

Comment:
I feel like this should be part of `template-az-cli-login.yaml` WDYT ?

Reply:
sounds good I'll switch it over

Reply:
I need the azure subscription I pull down as part of make secrets which I have to login before. Maybe I can just make a separate pipeline and only call it once instead of with every stage

Reply:
I couldn't there as I hadn't pulled it down with make secrets, but I'm now setting it right after

Reply:
oh i see, i'll resolve my comment then